staging/src/k8s.io/kubectl/pkg/cmd/diff/diff.go

```diff
@@ -741,6 +787,40 @@ func (o *DiffOptions) Run() error {
 		return err
 	})
 
+	if o.applySet != nil {
+		validationDirective := "ignore"
+		if err := o.applySet.BeforeApply(infos, cmdutil.DryRunServer, validationDirective); err != nil {
+			return err
+		}
+	}
+
+	if o.applySet != nil {
+		prunedObjects, err := o.applySet.FindObjectsToPrune(ctx, o.DynamicClient, o.tracker.visitedUids)
+		if err != nil {
+			klog.Warningf("pruning failed and could not be evaluated: %v", err)
```

Comment:
this sounds like we tried to prune... maybe something more specific to the "find" step we just attempted?

Reply:
Agreed.  It was copied from the "legacy" prune code below, but I polished it per your suggestion